lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -1378,50 +1380,66 @@ private static function sort_start_ascending( $a, $b ) {
 	}
 
 	/**
-	 * Given an attribute name, return its value as set in $lexical_updates.
+	 * Return the enqueued value for a given attribute, if one exists.
+	 *
+	 * Enqueued updates can take different data types:
+	 *  - If an update is enqueued and is boolean, the return will be `true`
+	 *  - If an update is otherwise enqueued, the return will be the string value of that update.
+	 *  - If an attribute is enqueued to be removed, the return will be `null` to indicate that.
+	 *  - If no updates are enqueued, the return will be `false` to differentiate from "removed."
 	 *
 	 * @since 6.2.0
 	 *
-	 * @param string $name The attribute name.
-	 * @return string|true|null Value of attribute or `null` if not available.
-	 *                          Boolean attributes return `true`.
+	 * @param string $comparable_name The attribute name in its comparable form.
+	 * @return string|boolean|null Value of enqueued update if present, otherwise false.
 	 */
-	private function extract_attribute_value_from_lexical_update( $name ) {
-		$comparable = strtolower( trim( $name ) );
-
-		if ( ! isset( $this->lexical_updates[ $comparable ] ) ) {
-			return null;
+	private function get_enqueued_attribute_value( $comparable_name ) {
```

Comment:
I love this change

Reply:
Actually... it confuses me how `add_class` etc updates are not considered "enqueued" by this method. It may be too much but here's an idea – what if we renamed `lexical_updates` to `enqueued_updates`?

Reply:
Renaming `lexical_updates` to `enqueued_updates` works for me. (I'd do it in a follow-up PR though, this one has grown big enough already 😬)

Reply:
I'm not bothered by the fact that class-builder updates aren't considered here. in fact there's a circularity problem if we make it work that way, because `class_updates_to_attribute_updates` needs to read this method.

as a private method it's not supposed to be exposed either. we can add a note about `class` being special.

as per `enqueued_updates` vs. `lexical_updates` I don't know. I don't see much win but I don't see a loss. the difference is that when we renamed it to `lexical_updates` it was meant to better reflect that these are textual diffs being applies without any other semantics. `enqueued_updates` somewhat hints that there might be more structure or semantic value associated with the updates when in fact there aren't and this whole system _so far_ has been built around the idea that they are intentionally dumb (well, apart from reprising them in this PR 🙃).

Reply:
I'm fine either way (and more than happy to defer the rename 😄)